c2corg_api/views/user.py

```diff
@@ -288,6 +296,146 @@ def post(self):
         return {}
 
 
+class UpdateAccountSchema(colander.MappingSchema):
+    email = colander.SchemaNode(
+            colander.String(),
+            missing=colander.drop,
+            validator=colander.All(
+                colander.Email(),
+                colander.Function(
+                    partial(available_user_attribute, 'email'),
+                    'Already used email'
+                )
+            ))
+    toponame = colander.SchemaNode(
+            colander.String(),
+            missing=colander.drop,
+            validator=colander.Length(min=3))
+    forumname = colander.SchemaNode(
+            colander.String(),
+            missing=colander.drop,
+            validator=colander.All(
+                colander.Length(min=3),
+                colander.Function(
+                    partial(available_user_attribute, 'forum_username'),
+                    'Already used forum name'
+                )
+            ))
+    currentpassword = colander.SchemaNode(
+            colander.String(encoding=ENCODING),
+            validator=colander.Length(min=3))
+    newpassword = colander.SchemaNode(
+            colander.String(encoding=ENCODING),
+            missing=colander.drop,
+            validator=colander.Length(min=3))
+
+
+@resource(path='/users/account', cors_policy=cors_policy)
+class UserAccountRest(object):
+    updateschema = UpdateAccountSchema()
+
+    def __init__(self, request):
+        self.request = request
+
+    def get_user(self):
+        userid = self.request.authenticated_userid
+        return DBSession.query(User).get(userid)
+
+    @restricted_view(renderer='json', http_cache=0)
+    def get(self):
+        user = self.get_user()
+        return {
```

Comment:
> @asaunier, what was decided for User.user_name? Can this be shown? 

I don't know. @gberaudo has attended a meeting with the client about the forum. I haven't. I remember they complained that the login appeared in the forum instead of the "pseudo" (forumname?). What has been decided with the client about this?

> The problem is that name (the old topo_name) is not a required field, thus it can be empty.

??
As far as I know the name/topo_name should be required since we need it as a title in the user profiles, in all guidebook pages where users are mentioned (history, outing participants, etc.).
Should I ask the client for a confirmation about how login/forum name/topo name should be used?